File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    import heapq
    answer = 0
    dic = {}
    for i in tangerine:
        if i in dic:
            dic[i] += 1
        else:
            dic[i] = 1
    lst = []
    for value in dic.values():
        heapq.heappush(lst,-1*value)

    while k>0 and lst:
        now = heapq.heappop(lst)
        if k < -1*now:
            answer += 1
            break
        k += now
        answer += 1

    
    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;

int solution(int k, vector<int> tangerine) {
    int answer = 0;
    sort(tangerine.begin(),tangerine.end());
    vector<int> check;
    int count = 0;
    int before = tangerine[0];
    for(auto now : tangerine){
        if(now==before){
            count++;
        }
        else{
            check.push_back(count);
            count = 1;
            before = now;
        }
    }
    check.push_back(count);
    sort(check.begin(),check.end(),greater());
    for(auto now : check){
        k-=now;
        answer++;
        if(k<=0)break;
    }
    return answer;
}